src/rxn_aa_mapper/model.py

```diff
@@ -14,28 +66,48 @@ class EnzymaticReactionLightningModule(pl.LightningModule):
     def __init__(
         self,
         model_args: Dict[str, Any],
-        model_architecture: Dict[str, Any],
     ) -> None:
         """
         Construct an EnzymaticReaction lightning module.
 
         Args:
             model_args: a dictionary object containing all the necessary arguments for the model creation.
-            model_architecture: a dictionary containing all the information related to the architecture of the model.
         """
         super().__init__()
         self.model_args = model_args
+        self.model_architecture = model_args.get("architecture", {})
+
         resume = self.model_args.get("resume-training", False)
         if resume:
             logger.info(
                 f"resume the training from the checkpoint : {self.model_args['model']}"
             )
             self.model = AutoModelForMaskedLM.from_pretrained(model_args["model"])
         else:
-            self.config = AutoConfig.from_pretrained(
-                model_args["model"], **model_architecture
-            )
-            self.model = AutoModelForMaskedLM.from_config(self.config)
+            if "albert" in model_args["model"].lower():
```

Comment:
I'm wondering whether it would be better to be explicitly and having a dictionary mapping specific strings to a class 🤔 .
Let's do the following, we keep it as is, but please open an issue about it so we can keep thinking on how to improve this.